src/clinical/service-worker-thread/tasks.ts

```diff
@@ -269,23 +276,18 @@ function extractEntityDataFromDonors(
     });
   });
 
-  const completionStats: CompletionRecord[] = donors
-    .map(({ completionStats, donorId, specimens }): CompletionRecord | undefined => {
-      let completionRecord =
-        completionStats && donorId ? { ...completionStats, donorId } : undefined;
-
-      // Update Completion Stats to display Normal/Tumour stats
-      if (completionRecord && completionRecord.coreCompletion?.specimens > 0) {
-        const donorSpecimenData = specimens.filter(dnaSampleFilter);
-
-        completionRecord = {
-          ...completionRecord,
-          coreCompletion: {
-            ...completionRecord.coreCompletion,
-            ...calculateSpecimenCompletionStats(donorSpecimenData),
-          },
-        };
-      }
+  const completionStats: CompletionDisplayRecord[] = donors
+    .map(({ completionStats, donorId, specimens: donorSpecimenRecords }):
+      | CompletionDisplayRecord
+      | undefined => {
```

Comment:
i'm unclear what is happening here.
`donors` looks like it's type is working ok. 
but the elements of donors are assigned new type annotations.
ye typescript gods @joneubank any thoughts?

separate point, in general
stylistically I'd make sure whatever you're mapping on is typed before the map so there isn't an ugly inline map type annotation.

Reply:
I'm reading too much into it I think
you can remove the `| CompletionDisplayRecord | undefined` 
those types are inferred correctly from the `donors` type `Donor[]` already

you are also returning the correct type of `CompletionDisplayRecord[]`

so this is minor cleanup

Reply:
Yes you are correct thank you!

Reply:
Updated